CMakeLists.txt

```diff
@@ -67,13 +76,16 @@ else()
     add_library(Savitar SHARED ${savitar_SRCS})
 endif()
 
-target_link_libraries(Savitar PUBLIC pthread pugixml)
+if(BUILD_WITH_THREADS)
+    target_link_libraries(Savitar PUBLIC pthread pugixml)
```

Comment:
You can use `CMAKE_THREAD_LIBS_INIT`. Example: `target_link_libraries(Savitar PUBLIC CMAKE_THREAD_LIBS_INIT pugixml)`. See https://cmake.org/cmake/help/v3.10/module/FindThreads.html